File: series2/series2_handout/multistep/plot_convergence.py
from numpy import *
from pylab import *
from os.path import isfile

errors_euler= loadtxt("errors.txt")
resolutions = loadtxt("resolutions.txt")

loglog(resolutions, errors_euler, label="FE")
xlabel("resolutions")
ylabel("errors")
grid()

if isfile("errors_ab2.txt"):
	errors_ab2 = loadtxt("errors_ab2.txt")
	loglog(resolutions, errors_ab2, label="AB2")

legend()
show()

File: series2/series2_code/series2_code/multistep/plot_convergence.m
clc
close all
clear all

FolderName = 'build';

errors_euler= load([FolderName,'/errors.txt']);
resolutions = load([FolderName,'/resolutions.txt']);

loglog(resolutions, errors_euler);
xlabel('resolutions');
ylabel('errors');

if exist([FolderName,'/errors_ab2.txt'], 'file') == 2 % exists and is a file
    hold on
    errors_ab2 = load([FolderName,'/errors_ab2.txt']);
    loglog(resolutions, errors_ab2);
    legend('euler', 'ab2');
else
    legend('euler');
end
grid minor